lua/lspconfig/powershell_es.lua

```diff
@@ -4,20 +4,27 @@ local util = require 'lspconfig/util'
 local server_name = 'powershell_es'
 local temp_path = vim.fn.stdpath 'cache'
 
-local function make_cmd(bundle_path)
-  if bundle_path ~= nil then
+local function make_cmd(new_config)
+  local command = ""
+  if new_config.bundle_path ~= nil then
     local command_fmt =
       [[%s/PowerShellEditorServices/Start-EditorServices.ps1 -BundledModulesPath %s -LogPath %s/powershell_es.log -SessionDetailsPath %s/powershell_es.session.json -FeatureFlags @() -AdditionalModules @() -HostName nvim -HostProfileId 0 -HostVersion 1.0.0 -Stdio -LogLevel Normal]]
-    local command = command_fmt:format(bundle_path, bundle_path, temp_path, temp_path)
-    return { 'pwsh', '-NoLogo', '-NoProfile', '-Command', command }
+    command = command_fmt:format(new_config.bundle_path, new_config.bundle_path, temp_path, temp_path)
   end
+  if new_config.shell == nil then
+    -- defaulting to PowerShell Core
+    new_config.shell = 'pwsh'
+  end
+  new_config.cmd = { new_config.shell, '-NoLogo', '-NoProfile', '-Command', command }
 end
 
 configs[server_name] = {
   default_config = {
     on_new_config = function(new_config, _)
-      local bundle_path = new_config.bundle_path
-      new_config.cmd = make_cmd(bundle_path)
+      if new_config.cmd == nil then
+        -- Only create a new cmd if it isn't specified in the new config
+        make_cmd(new_config)
```

Comment:
I prefer making this assignment explicit, return cmd like before and set new_config.cmd = cmd.